### DISCUSSION TITLE
Using ATIP to identify OSM issues where routes are not connected

### DISCUSSION THREAD

**Discussion by Robinlovelace** [acteng/atip#87] [Ideas]
As illustrated below for part of Leeds:

![image](https://user-images.githubusercontent.com/1825120/222837816-2581a8d1-defd-4c26-8f00-af02216da98c.png)

Forgiving routing tools can probably route here but it shows the power of routing on-the-fly to debug OSM, just a discovery so thought I'd share.

Source: https://atip.uk/scheme.html?authority=West%20Yorkshire%20Combined%20Authority#17.98/53.800647/-1.54746

**Comment by Robinlovelace** [comment]
Interestingly, it affects at least one other routing engine that is tolerant to poor OSM edits: CycleStreets. FYI @mvl22:

![image](https://user-images.githubusercontent.com/1825120/222838412-d69dc35c-eb1f-4fef-b6aa-f7fca55dad7e.png)

Source: https://www.cyclestreets.net/journey/88795054/

**Reply by Robinlovelace** [reply]
OK the OSM is way off, down as under construction: https://www.openstreetmap.org/edit?way=1092013872#map=20/53.80087/-1.54721

**Reply by Robinlovelace** [reply]
OK seems it genuinely IS closed, I missed this: https://api-gb.one.network/downloads/tm/1180/ttrn-order-leonardo_21dbe92aca.pdf 

Source: https://one.network/uk/leeds

**Reply by Robinlovelace** [reply]
Questions arising: should ATIP allow routing on paths under construction or that are blocked? Should they be 'fault tolerant' and allow jumping over small gaps in the network? I vote YES to both.

**Reply by mvl22** [reply]
No they shouldn’t.

Firstly an area under construction in OSM tends to be for a significant period. Adding the construction tag is recommended against for anything short.

But the main point is that an area under construction for a significant period is likely to mean some major change to an area. E.g. a pedestrianisation scheme. Or a change to layouts and directions etc. These future changes can’t be present in the data yet. So it is just as inaccurate to assume the pre-construction layout will have the same layout after.

**Reply by dabreegster** [reply]
I'm afraid there's some miscommunication. The route tool in ATIP has a very different purpose than a routing engine intended for humans to follow the route. I added https://github.com/dabreegster/route_snapper/blob/main/user_guide.md#routing-caveats to clarify.

Robin, in ATIP, I agree that the more appropriate thing would be to ignore `highway=construction` and also one-ways. I'm not confident either one is working that way right now; https://github.com/dabreegster/route_snapper/issues/21 is a placeholder to do something else.

Also, the confusion about what the ATIP routes "mean" will be greatly improved when we start some of the next proposed phases of the tool: showing existing traffic restrictions, historic speeds / traffic volumes, and lane configuration, all as a hint for the user to understand what they'll need to change their to deliver something LTN120 compliant.

**Reply by Robinlovelace** [reply]
Yes ATIP routes are not designed to be followed, they are designed to enable people to think about where new investment could go. So the more the merrier is reasonable. Surely `highway=construction` should be included in range of routes that should be considered when thinking about where to invest, and one-way streets (which could receive high quality contraflow).

**Reply by Robinlovelace** [reply]
> No they shouldn’t.
> 
> Firstly an area under construction in OSM tends to be for a significant period. Adding the construction tag is recommended against for anything short.
> 
> But the main point is that an area under construction for a significant period is likely to mean some major change to an area. E.g. a pedestrianisation scheme. Or a change to layouts and directions etc. These future changes can’t be present in the data yet. So it is just as inaccurate to assume the pre-construction layout will have the same layout after.

Including ways under construction does not assume they will have the same layout after. It just means that people can sketch new infrastructure along roads that have not yet been completed if they want. This is definitely not a journey planning tool, and that may have caused some confusion. 

I thought for a moment I had found an issue with CycleStreets routing but subsequently found that indeed it correctly re-routes around the roadworks. For ATIP however, people should be able to sketch routes along that and other ways IMO, the wider the range of ways that people can sketch routes on the better for this rather niche use case.
